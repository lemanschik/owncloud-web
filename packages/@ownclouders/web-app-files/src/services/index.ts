export * from './archiver'
export { default as Registry } from './registry'
