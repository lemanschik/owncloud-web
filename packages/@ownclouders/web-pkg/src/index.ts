export * from './composables'
export * from './configuration'
export * from './helpers'
export * from './http'
export * from './observer'
export * from './services'
export * from './utils'
