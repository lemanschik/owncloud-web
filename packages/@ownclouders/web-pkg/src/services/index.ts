export * from './cache'
export * from './client'
export * from './eventBus'
export * from './loadingService'
export * from './preview'
