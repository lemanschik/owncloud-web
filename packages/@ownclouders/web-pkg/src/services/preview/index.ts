export * from './previewService'
export * from './types'
