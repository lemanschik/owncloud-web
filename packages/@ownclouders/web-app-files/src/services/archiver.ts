// Workaround https://github.com/npm/node-semver/issues/381
import major from 'semver/functions/major'
import rcompare from 'semver/functions/rcompare'

import { RuntimeError } from 'web-runtime/src/container/error'
import { ClientService } from 'web-pkg/src/services'
import { urlJoin } from 'web-client/src/utils'
import { configurationManager } from 'web-pkg/src/configuration'
import { triggerDownloadWithFilename } from 'web-pkg/src/helpers'
/**
 * Archiver struct within the capabilities as defined in reva
 * @see https://github.com/cs3org/reva/blob/41d5a6858c2200a61736d2c165e551b9785000d1/internal/http/services/owncloud/ocs/data/capabilities.go#L105
 */
export interface ArchiverCapability {
  enabled: boolean
  version: string // version is just a major version, e.g. `v2`
  formats: string[]
  // eslint-disable-next-line camelcase
  archiver_url: string
}

interface TriggerDownloadOptions {
  dir?: string
  files?: string[]
  fileIds?: string[]
  downloadSecret?: string
  publicToken?: string
  publicLinkPassword?: string
}

export class ArchiverService {
  clientService: ClientService
  serverUrl: string
  capability?: ArchiverCapability

  constructor(
    clientService: ClientService,
    serverUrl: string,
    archiverCapabilities: ArchiverCapability[] = []
  ) {
    this.clientService = clientService
    this.serverUrl = serverUrl
    const archivers = archiverCapabilities
      .filter((a) => a.enabled)
      .sort((a1, a2) => rcompare(a1.version, a2.version))
    this.capability = archivers.length ? archivers[0] : null
  }

  public get available(): boolean {
    return !!this.capability?.version
  }

  public get fileIdsSupported(): boolean {
    return major(this.capability?.version) >= 2
  }

  public async triggerDownload(options: TriggerDownloadOptions): Promise<string> {
    if (!this.available) {
      throw new RuntimeError('no archiver available')
    }

    if ((options.fileIds?.length || 0) + (options.files?.length || 0) === 0) {
      throw new RuntimeError('requested archive with empty list of resources')
    }

    const downloadUrl = this.buildDownloadUrl({ ...options })
    if (!downloadUrl) {
      throw new RuntimeError('download url could not be built')
    }

    const url = options.publicToken
      ? downloadUrl
      : await this.clientService.owncloudSdk.signUrl(downloadUrl)

    try {
      const response = await this.clientService.httpUnAuthenticated.get(url, {
        headers: {
          ...(!!options.publicLinkPassword && {
            Authorization:
              'Basic ' +
              Buffer.from(['public', options.publicLinkPassword].join(':')).toString('base64')
          })
        },
        responseType: 'arraybuffer'
      })

      const blob = new Blob([response.data], { type: 'application/octet-stream' })
      const objectUrl = URL.createObjectURL(blob)
      const fileName = this.getFileNameFromResponseHeaders(response.headers)
      triggerDownloadWithFilename(objectUrl, fileName)
      return url
    } catch (e) {
      throw new RuntimeError('archive could not be fetched')
    }
  }

  private buildDownloadUrl(options: TriggerDownloadOptions): string {
    const queryParams = []
    if (options.publicToken) {
      queryParams.push(`public-token=${options.publicToken}`)
    }

    const majorVersion = major(this.capability.version)
    switch (majorVersion) {
      case 2: {
        queryParams.push(...options.fileIds.map((id) => `id=${id}`))
        return this.url + '?' + queryParams.join('&')
      }
      case 1: {
        // see https://github.com/owncloud/core/blob/e285879a8a79e692497937ebf340bc6b9c925b4f/apps/files/js/files.js#L315 for reference
        // classic ui does a check whether the download started. not implemented here (yet?).
        const downloadStartSecret = Math.random().toString(36).substring(2)
        queryParams.push(
          `dir=${encodeURIComponent(options.dir)}`,
          ...options.files.map((name) => `files[]=${encodeURIComponent(name)}`),
          `downloadStartSecret=${downloadStartSecret}`
        )
        return this.url + '?' + queryParams.join('&')
      }
      default: {
        return undefined
      }
    }
  }

  private get url(): string {
    if (!this.available) {
      throw new RuntimeError('no archiver available')
    }
    if (/^https?:\/\//i.test(this.capability.archiver_url)) {
      return this.capability.archiver_url
    }
    return urlJoin(configurationManager.serverUrl, this.capability.archiver_url)
  }

  private getFileNameFromResponseHeaders(headers) {
    const fileName = headers['content-disposition']?.split('"')[1]
    return decodeURI(fileName)
  }
}
