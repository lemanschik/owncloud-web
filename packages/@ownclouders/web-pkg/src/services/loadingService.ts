import * as uuid from 'uuid'
import { eventBus } from 'web-pkg'
import { debounce } from 'lodash-es'

export enum LoadingEventTopics {
  add = 'loading-service.add',
  remove = 'loading-service.remove',
  setProgress = 'loading-service.set-progress'
}

export interface LoadingTaskState {
  total: number
  current: number
}

export interface LoadingTask {
  id: string
  active: boolean
  state?: LoadingTaskState
}

export interface LoadingTaskCallbackArguments {
  setProgress: (args: LoadingTaskState) => void
}

// time until a loading task is being set active
const DEFAULT_DEBOUNCE_TIME = 200

export class LoadingService {
  private tasks: LoadingTask[] = []

  public get isLoading(): boolean {
    return this.tasks.some((e) => e.active)
  }

  /**
   * Get the current progress from 0 to 100.
   * Returns null if at least one task is indeterminate.
   */
  public get currentProgress(): number | null {
    if (this.tasks.some((e) => !e.state && e.active)) {
      return null
    }

    const tasks = this.tasks.filter((e) => !!e.state && e.active)
    if (!tasks.length) {
      return null
    }

    const progress = tasks.reduce((acc, task) => {
      acc += task.state.current / task.state.total
      return acc
    }, 0)

    return Math.round((progress / tasks.length) * 100)
  }

  public addTask<T>(
    callback: ({ setProgress }: LoadingTaskCallbackArguments) => Promise<T>,
    {
      debounceTime = DEFAULT_DEBOUNCE_TIME,
      indeterminate = true
    }: { debounceTime?: number; indeterminate?: boolean } = {}
  ): Promise<T> {
    const task = {
      id: uuid.v4(),
      active: false,
      ...(!indeterminate && { state: { total: 0, current: 0 } })
    }
    this.tasks.push(task)

    const debounced = debounce(() => {
      task.active = true
      eventBus.publish(LoadingEventTopics.add)
    }, debounceTime)
    debounced()

    const setProgress = ({ total, current }: LoadingTaskState) => {
      if (!indeterminate) {
        this.setProgress({ task, total, current })
      }
    }

    return callback({ setProgress }).finally(() => {
      this.removeTask(task.id)
    })
  }

  private removeTask(id: string): void {
    this.tasks = this.tasks.filter((e) => e.id !== id)
    eventBus.publish(LoadingEventTopics.remove)
  }

  private setProgress({ task, total, current }): void {
    if (!task.state) {
      task.state = { total: 0, current: 0 }
    }
    task.state.total = total
    task.state.current = current
    eventBus.publish(LoadingEventTopics.setProgress)
  }
}
