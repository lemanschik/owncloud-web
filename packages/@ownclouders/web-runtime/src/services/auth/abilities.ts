import { SubjectRawRule } from '@casl/ability'
import { AbilityActions, AbilitySubjects } from 'web-pkg'

export const getAbilities = (
  permissions: string[]
): SubjectRawRule<AbilityActions, AbilitySubjects, any>[] => {
  const abilities: Record<string, SubjectRawRule<AbilityActions, AbilitySubjects, any>[]> = {
    'account-management.all': [
      { action: 'create-all', subject: 'Account' },
      { action: 'delete-all', subject: 'Account' },
      { action: 'read-all', subject: 'Account' },
      { action: 'update-all', subject: 'Account' }
    ],
    'group-management.all': [
      { action: 'create-all', subject: 'Group' },
      { action: 'delete-all', subject: 'Group' },
      { action: 'read-all', subject: 'Group' },
      { action: 'update-all', subject: 'Group' }
    ],
    'language-readwrite.all': [
      { action: 'read-all', subject: 'Language' },
      { action: 'update-all', subject: 'Language' }
    ],
    'change-logo.all': [{ action: 'update-all', subject: 'Logo' }],
    'PublicLink.Write.all': [{ action: 'create-all', subject: 'PublicLink' }],
    'role-management.all': [
      { action: 'create-all', subject: 'Role' },
      { action: 'delete-all', subject: 'Role' },
      { action: 'read-all', subject: 'Role' },
      { action: 'update-all', subject: 'Role' }
    ],
    'settings-management.all': [
      { action: 'read-all', subject: 'Setting' },
      { action: 'update-all', subject: 'Setting' }
    ],
    'create-space.all': [{ action: 'create-all', subject: 'Space' }],
    'Drive.ReadWriteEnabled.all': [
      { action: 'delete-all', subject: 'Space' },
      { action: 'update-all', subject: 'Space' }
    ],
    'list-all-spaces.all': [{ action: 'read-all', subject: 'Space' }],
    'Drive.ReadWriteQuota.Project.all': [{ action: 'set-quota-all', subject: 'Space' }]
  }

  return Object.keys(abilities).reduce((acc, permission) => {
    if (permissions.includes(permission)) {
      acc.push(...abilities[permission])
    }
    return acc
  }, [])
}
