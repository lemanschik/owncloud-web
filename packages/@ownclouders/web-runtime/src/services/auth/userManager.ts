import {
  Log,
  WebStorageStateStore,
  UserManager as OidcUserManager,
  UserManagerSettings
} from 'oidc-client-ts'
import { buildUrl } from '../../router'
import { getAbilities } from './abilities'
import { ConfigurationManager } from 'web-pkg/src/configuration'
import { ClientService } from 'web-pkg/src/services'
import { Store } from 'vuex'
import isEmpty from 'lodash-es/isEmpty'
import { Ability } from 'web-pkg/src/utils'
import { Language } from 'vue3-gettext'
import { setCurrentLanguage } from 'web-runtime/src/helpers/language'

const postLoginRedirectUrlKey = 'oc.postLoginRedirectUrl'
type UnloadReason = 'authError' | 'logout'

export interface UserManagerOptions {
  clientService: ClientService
  configurationManager: ConfigurationManager
  store: Store<any>
  ability: Ability
  language: Language
}

export class UserManager extends OidcUserManager {
  private readonly storePrefix
  private clientService: ClientService
  private configurationManager: ConfigurationManager
  private store: Store<any>
  private updateAccessTokenPromise: Promise<void> | null
  private _unloadReason: UnloadReason
  private ability: Ability
  private language: Language

  public areEventHandlersRegistered: boolean

  constructor(options: UserManagerOptions) {
    const storePrefix = 'oc_oAuth.'
    const userStore = new WebStorageStateStore({
      prefix: storePrefix,
      store: sessionStorage
    })
    const openIdConfig: UserManagerSettings = {
      userStore,
      redirect_uri: buildUrl('/oidc-callback.html'),
      silent_redirect_uri: buildUrl('/oidc-silent-redirect.html'),

      response_mode: 'query',
      response_type: 'code', // "code" triggers auth code grant flow

      post_logout_redirect_uri: buildUrl('/'),
      accessTokenExpiringNotificationTimeInSeconds: 10,
      authority: '',
      client_id: ''
    }

    if (options.configurationManager.isOIDC) {
      Object.assign(openIdConfig, {
        scope: 'openid profile',
        loadUserInfo: true,
        ...options.configurationManager.oidc
      })
    } else if (options.configurationManager.isOAuth2) {
      const oAuth2 = options.configurationManager.oAuth2
      Object.assign(openIdConfig, {
        authority: oAuth2.url,
        client_id: oAuth2.clientId,
        ...(oAuth2.clientSecret && {
          client_authentication: 'client_secret_basic',
          client_secret: oAuth2.clientSecret
        }),

        scope: 'profile',
        loadUserInfo: false,
        metadata: {
          issuer: oAuth2.url,
          authorization_endpoint: oAuth2.authUrl,
          token_endpoint: oAuth2.url,
          userinfo_endpoint: ''
        }
      })
    }

    Log.setLogger(console)
    Log.setLevel(Log.INFO)

    super(openIdConfig)
    this.storePrefix = storePrefix
    this.clientService = options.clientService
    this.configurationManager = options.configurationManager
    this.store = options.store
    this.ability = options.ability
    this.language = options.language
  }

  /**
   * Looks up the access token of an already loaded user without enforcing a signin if no user exists.
   *
   * @return (string|null)
   */
  async getAccessToken(): Promise<string | null> {
    const user = await this.getUser()
    return user?.access_token
  }

  async removeUser(unloadReason: UnloadReason = 'logout') {
    this._unloadReason = unloadReason
    await super.removeUser()
  }

  get unloadReason(): UnloadReason {
    return this._unloadReason
  }

  getAndClearPostLoginRedirectUrl(): string {
    const url = sessionStorage.getItem(postLoginRedirectUrlKey) || '/'
    sessionStorage.removeItem(postLoginRedirectUrlKey)
    return url
  }

  setPostLoginRedirectUrl(url?: string): void {
    if (url) {
      sessionStorage.setItem(postLoginRedirectUrlKey, url)
    } else {
      sessionStorage.removeItem(postLoginRedirectUrlKey)
    }
  }

  updateContext(accessToken: string, fetchUserData: boolean): Promise<void> {
    const userKnown = !!this.store.getters.user.id
    const accessTokenChanged = this.store.getters['runtime/auth/accessToken'] !== accessToken
    if (!accessTokenChanged) {
      return this.updateAccessTokenPromise
    }
    this.store.commit('runtime/auth/SET_ACCESS_TOKEN', accessToken)

    this.updateAccessTokenPromise = (async () => {
      if (!fetchUserData) {
        this.store.commit('runtime/auth/SET_IDP_CONTEXT_READY', true)
        return
      }

      const settings = await this.fetchSettings()
      setCurrentLanguage({
        language: this.language,
        languageSetting: settings?.find((s) => s.identifier.setting === 'language'),
        user: this.store.getters.user
      })
      this.initializeOwnCloudSdk(accessToken)

      if (!userKnown) {
        await this.fetchUserInfo()
        await this.updateUserAbilities(this.store.getters.user)
        this.store.commit('runtime/auth/SET_USER_CONTEXT_READY', true)
      }
    })()
    return this.updateAccessTokenPromise
  }

  private initializeOwnCloudSdk(accessToken: string): void {
    const options = {
      baseUrl: this.configurationManager.serverUrl,
      auth: {
        bearer: accessToken
      },
      headers: {
        'Accept-Language': this.language.current
      }
    }
    if (this.store.getters.user.id) {
      ;(options as any).userInfo = {
        id: this.store.getters.user.id,
        'display-name': this.store.getters.user.displayname,
        email: this.store.getters.user.email
      }
    }

    this.clientService.owncloudSdk.init(options)
  }

  private async fetchUserInfo(): Promise<void> {
    const [login] = await Promise.all([
      this.clientService.owncloudSdk.getCurrentUser(),
      this.fetchCapabilities()
    ])

    let role, graphUser, userGroups

    // start the network request here and let it run in parallel to the next calls,
    // until awaiting the promise execution further below
    const userPromise = this.clientService.owncloudSdk.users.getUser(login.id)

    if (this.store.getters.capabilities.spaces?.enabled) {
      const graphClient = this.clientService.graphAuthenticated
      let roles
      ;[graphUser, roles] = await Promise.all([graphClient.users.getMe(), this.fetchRoles()])
      this.store.commit('SET_ROLES', roles)

      role = await this.fetchRole({ graphUser, roles })
    } else {
      userGroups = await this.clientService.owncloudSdk.users.getUserGroups(login.id)
    }

    const user = await userPromise

    this.store.commit('SET_USER', {
      id: login.id,
      uuid: graphUser?.data?.id || '',
      username: login.username || login.id,
      displayname: user.displayname || login['display-name'],
      email: login?.email || user?.email || '',
      groups: graphUser?.data?.memberOf || userGroups || [],
      role,
      language: login?.language
    })

    if (login.language) {
      // language is saved in user object for oC10
      setCurrentLanguage({ language: this.language, user: login })
    }

    if (!this.store.getters.capabilities.spaces?.enabled && user.quota) {
      this.store.commit('SET_QUOTA', user.quota)
    }
  }

  private async fetchRoles(): Promise<any> {
    const httpClient = this.clientService.httpAuthenticated
    try {
      const {
        data: { bundles: roles }
      } = await httpClient.post('/api/v0/settings/roles-list', {})
      return roles
    } catch (e) {
      console.error(e)
      return []
    }
  }

  private async fetchSettings(): Promise<any> {
    const httpClient = this.clientService.httpAuthenticated
    try {
      const {
        data: { values }
      } = await httpClient.post('/api/v0/settings/values-list', { account_uuid: 'me' })
      return values
    } catch (e) {
      console.error(e)
      return null
    }
  }

  private async fetchRole({ graphUser, roles }): Promise<any> {
    const httpClient = this.clientService.httpAuthenticated
    const userAssignmentResponse = await httpClient.post('/api/v0/settings/assignments-list', {
      account_uuid: graphUser.data.id
    })
    const assignments = userAssignmentResponse.data?.assignments
    const roleAssignment = assignments.find((assignment) => 'roleId' in assignment)
    return roleAssignment ? roles.find((role) => role.id === roleAssignment.roleId) : null
  }

  private async fetchCapabilities(): Promise<void> {
    if (!isEmpty(this.store.getters.capabilities)) {
      return
    }

    const capabilities = await this.clientService.ocsUserContext.getCapabilities()

    this.store.commit('SET_CAPABILITIES', capabilities)
  }

  private async fetchPermissions({ user }): Promise<any> {
    const httpClient = this.clientService.httpAuthenticated
    const oC10DefaultPermissions = ['PublicLink.Write.all']
    try {
      const {
        data: { permissions }
      } = await httpClient.post('/api/v0/settings/permissions-list', { account_uuid: user.uuid })
      return permissions || oC10DefaultPermissions
    } catch (e) {
      console.error(e)
      return oC10DefaultPermissions
    }
  }

  private async updateUserAbilities(user) {
    const permissions = await this.fetchPermissions({ user })
    const abilities = getAbilities(permissions)
    this.ability.update(abilities)
  }
}
