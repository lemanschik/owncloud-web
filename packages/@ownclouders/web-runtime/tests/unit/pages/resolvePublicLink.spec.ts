import ResolvePublicLink from '../../../src/pages/resolvePublicLink.vue'
import {
  defaultPlugins,
  defaultComponentMocks,
  createStore,
  shallowMount,
  defaultStoreMockOptions
} from 'web-test-helpers'
import { mockDeep } from 'jest-mock-extended'
import { ClientService } from 'web-pkg'
import { Resource } from 'web-client'
import { authService } from 'web-runtime/src/services/auth'

jest.mock('web-runtime/src/services/auth')
const selectors = {
  cardFooter: '.oc-card-footer',
  ocSpinnerStub: 'oc-spinner-stub',
  submitButton: '.oc-login-authorize-button'
}

describe('resolvePublicLink', () => {
  it('should display the configuration theme general slogan as the login card footer', () => {
    const { wrapper } = getWrapper()
    const slogan = wrapper.find(selectors.cardFooter)
    expect(slogan.html()).toMatchSnapshot()
  })
  it('should display the loading spinner', () => {
    const { wrapper } = getWrapper({ passwordRequired: true })
    const loading = wrapper.find(selectors.ocSpinnerStub)
    expect(loading.exists()).toBeTruthy()
  })
  describe('password required form', () => {
    it('should display if password is required', async () => {
      const { wrapper } = getWrapper({ passwordRequired: true })
      await wrapper.vm.loadTokenInfoTask.last
      await wrapper.vm.isPasswordRequiredTask.last
      await wrapper.vm.$nextTick()
      expect(wrapper.find('form').html()).toMatchSnapshot()
    })
    describe('submit button', () => {
      it('should be set as disabled if "password" is empty', async () => {
        const { wrapper } = getWrapper({ passwordRequired: true })
        await wrapper.vm.loadTokenInfoTask.last
        await wrapper.vm.isPasswordRequiredTask.last
        await wrapper.vm.$nextTick()
        expect(wrapper.find(selectors.submitButton).attributes().disabled).toBe('true')
      })
      it('should be set as enabled if "password" is not empty', async () => {
        const { wrapper } = getWrapper({ passwordRequired: true })
        await wrapper.vm.loadTokenInfoTask.last
        await wrapper.vm.isPasswordRequiredTask.last
        await wrapper.vm.$nextTick()
        wrapper.vm.password = 'password'
        await wrapper.vm.$nextTick()
        expect(wrapper.find(selectors.submitButton).attributes().disabled).toBe('false')
      })
      it('should resolve the public link on click', async () => {
        const resolvePublicLinkSpy = jest.spyOn(authService, 'resolvePublicLink')
        const { wrapper } = getWrapper({ passwordRequired: true })
        await wrapper.vm.loadTokenInfoTask.last
        await wrapper.vm.isPasswordRequiredTask.last
        await wrapper.vm.$nextTick()
        wrapper.vm.password = 'password'
        await wrapper.vm.$nextTick()
        await wrapper.find(selectors.submitButton).trigger('submit')
        await wrapper.vm.resolvePublicLinkTask.last
        expect(resolvePublicLinkSpy).toHaveBeenCalled()
      })
    })
  })
})

function getWrapper({ passwordRequired = false } = {}) {
  const tokenInfo = { password_protected: passwordRequired ? 'true' : 'false' }
  const $clientService = mockDeep<ClientService>()
  $clientService.webdav.getFileInfo.mockResolvedValue(mockDeep<Resource>({ driveType: 'public' }))
  $clientService.owncloudSdk.shares.getUnprotectedTokenInfo.mockResolvedValue(tokenInfo)
  $clientService.owncloudSdk.shares.getProtectedTokenInfo.mockResolvedValue(tokenInfo)
  const mocks = { ...defaultComponentMocks(), $clientService }

  const storeOptions = defaultStoreMockOptions
  storeOptions.getters.capabilities.mockImplementation(() => ({
    files_sharing: { federation: { incoming: true, outgoing: true } }
  }))
  storeOptions.getters.configuration.mockReturnValue({
    currentTheme: { general: { slogan: 'Public link slogan' } }
  })
  const store = createStore(storeOptions)
  return {
    wrapper: shallowMount(ResolvePublicLink, {
      global: {
        plugins: [...defaultPlugins(), store],
        mocks
      }
    })
  }
}
