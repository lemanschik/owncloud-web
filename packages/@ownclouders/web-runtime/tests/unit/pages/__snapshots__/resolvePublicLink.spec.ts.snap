// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolvePublicLink password required form should display if password is required 1`] = `
<form>
  <div class="oc-card-header">
    <h2>
      <span>This resource is password-protected</span>
    </h2>
  </div>
  <div class="oc-card-body">
    <oc-text-input-stub class="oc-mb-s" clearbuttonaccessiblelabel="" clearbuttonenabled="false" disabled="false" fixmessageline="false" id="oc-textinput-2" label="Enter password for public link" modelvalue="" readonly="false" type="password"></oc-text-input-stub>
    <oc-button-stub appearance="filled" class="oc-login-authorize-button" disabled="true" gapsize="medium" justifycontent="center" size="medium" submit="submit" type="button" variation="primary"></oc-button-stub>
  </div>
</form>
`;

exports[`resolvePublicLink should display the configuration theme general slogan as the login card footer 1`] = `
<div class="oc-card-footer oc-pt-rm">
  <p>Public link slogan</p>
</div>
`;
